src/tequila/circuit/compiler.py

```diff
@@ -1297,3 +1315,83 @@ def compile_trotterized_gate(gate, compile_exponential_pauli: bool = False):
         return compile_exponential_pauli_gate(result)
     else:
         return result
+
+
+@compiler
+def compile_ry(gate: RotationGateImpl, controlled_rotation: bool = False) -> QCircuit:
+    """
+    Compile Ry gates into Rx and Rz.
+    Parameters
+    ----------
+    gate:
+        the gate.
+    controlled_rotation:
+        to determine if the gate break down need to be applied here
+
+    Returns
+    -------
+    QCircuit, the result of compilation.
+    """
+    if gate.name.lower() == "ry":
+
+        if not (gate.is_controlled() and controlled_rotation):
+
+            return Rz(target=gate.target, control=None, angle=-numpy.pi / 2) \
+                   + Rx(target=gate.target, control=gate.control, angle=gate.parameter) \
+                   + Rz(target=gate.target, control=None, angle=numpy.pi / 2)
+
+    return QCircuit.wrap_gate(gate)
+
+
+@compiler
+def compile_y(gate: QGateImpl) -> QCircuit:
+    """
+    Compile Y gates into X and Z.
+    Parameters
+    ----------
+    gate:
+        the gate.
+
+    Returns
+    -------
+    QCircuit, the result of compilation.
+    """
+    if gate.name.lower() == "y":
+
```

Comment:
Y gates can be parametrized with powers
e.g. tq.gates.Y(target=0, power=1.5) is Y^{1.5} 
that needs to be passed on to the decomposition (or at least asserted that the power is 1.0)

Reply:
could imagine that the power is just an additional factor for the Rx angle.
Relates to a lot of things that Georgios did, so he probably knows already

Reply:
Yes it's just an additional factor. But this decomposition will be off by a relative phase if Y-gate is controlled, so `Rx(target=gate.target, control=gate.control, angle=numpy.pi)` should be replaced with `X(target=gate.target, control=gate.control, power=gate.parameter)`

But in fact, with all compiler flags set to True, there is no need to handle power gates at this step and `compile_ry`, `compile_y`, `compile_ch` can be called as a final step to decompose plain Y, CY, Ry, CH gates.

Reply:
Great!
(Would still be good to add it, then it can in principle be used without the preceding steps by potential future applications)

Reply:
I have taken your suggestions and put the equivalency as `Y^a = Rz(-0.5pi) X^a Rz(0.5pi)`

Now consider the power parameter, and it does not depend on the other flags for the conversion to be complete.